### ISSUE TITLE
Helm Chart - IBM License operator

### ISSUE THREAD

**Issue by p9139010** [IBM/ibm-licensing-operator#760]
Hi,

Is there any helm chart available or in pipeline to have the license operator installed and managed more efficiently than shell scripts. It would help us in GitOps CD deployment on all clusters.

**Comment by dhpalan** [comment]
Yes, Please. +1
Deploying using shell scripts to a K8s cluster? We are not in a stone age!

**Comment by pgodowski** [comment]
> I don't have use cases either.
> I am merely suggesting that you provide us a standard solution.

Can you please clarify why Helm would be a better fit for your organization compared to OLM-based operator which is the first class citizen in the OpenShift and other Kubernetes distributions? We are not asking the customers to use shell scripts - we provide the mature implementation using Operators.


> You have here a huge collection of Helm charts for many IBM products. https://github.com/IBM/charts IBM ODM comes with a helm chart -> https://www.ibm.com/docs/en/odm/8.10?topic=production-installing-helm-release-odm
> Why can't `ibm-licensing-operator` have a helm chart?

While IBM has a collection of Helm charts, we have (larger) collection of OLM-based operators for IBM software. ODM also provides operator, so wondering whether you are using ODM? If so, what is your env and why Helm is better fit than operator?

**Comment by dhpalan** [comment]
To be honest, i wasn't aware of OLM. Literally searching on Google now.
And a bit shocked that helm is not a preferred choice anymore :)

I have been using HELM all the time. And of course I am using a helm chart to deploy ODM.

It is running on Azure Kubernetes Service (AKS).
I went with the [Offline installation](https://www.ibm.com/docs/en/cpfs?topic=software-offline-installation) for ibm-licensing-operator. And for convenience, wrote all those sed & kubectl commands into a shell script.

**Comment by pgodowski** [comment]
Thanks, I do understand better. On AKS, OLM is not deployed OOTB, and that is the reason why perhaps your org is not using it (yet?).

Operators are more than just application packaging. Operators provides the way to ensure the application lifecycle is managed e2e, including subscription for upgrades, automation of application configuration updates via registered custom controllers, and ensuring that application state is consistent.

Usually Helm is positioned as Day1 (basic install) operations only, while Operators are Day2  (upgrade, reconfiguration, backup, restore) operations.

https://operatorframework.io/what/

**Comment by dhpalan** [comment]
Yes. You are right. This has to be the reason. Moreover before AKS we were using a private Kubernetes cluster.
Will enquire if someone's already using Operators in my Org.

Sounds really interesting.
Thanks a lot for your time and the exchange.

**Comment by pgodowski** [comment]
Thank you. Please keep up posted, we will evaluate the outcome of your discussions and see how we can help.

**Comment by lorenzboguhn** [comment]
Hi, @p9139010 @dhpalan 
Not sure if this is still relevant for you but i just stumbled upon a helm chart added last month. https://github.com/IBM/ibm-licensing-operator/tree/latest-4.x/deploy/argo-cd/components/license-service/helm
Best Regards 
Lorenz
